addons/l10n_tr_nilvera_einvoice/models/account_edi_xml_ubl_tr.py

```diff
@@ -74,9 +80,16 @@ def _get_partner_address_vals(self, partner):
 
     def _get_partner_party_tax_scheme_vals_list(self, partner, role):
         # EXTENDS account.edi.xml.ubl_21
+        if (is_einvoice := partner.l10n_tr_nilvera_customer_status == "einvoice") and not partner.ref:
+            raise UserError(_("E-Invoice customers must have a tax office name in the partner reference field."))
+
         vals_list = super()._get_partner_party_tax_scheme_vals_list(partner, role)
         for vals in vals_list:
             vals.pop('registration_address_vals', None)
+            if is_einvoice:
+                vals["tax_scheme_vals"].update(
+                    {"id": "", "name": partner.ref}
+                )
```

Comment:
```suggestion
            if is_einvoice:
                vals["tax_scheme_vals"].update(
                    {
                        "id": "",
                        "name": partner.ref,
                    }
                )
```
Indentation + why are we adding an empty ID?

Reply:
I think there should be a better solution to this, but as we're short on time and the POs approved this, let's do it for now and keep in mind that we should investigate more later